training/performance/README.md

```diff
@@ -118,8 +131,25 @@ For example [Megatron-LM](https://github.com/NVIDIA/Megatron-LM) published the f
 | 530B  | 56.0% | 57.0% |
 | 1T    | 56.3% | 57.0% |
 
+As you can see, since Megatron-LM was using activation recomputation for these trainings, MFU < HFU.
+
 More recent H100+A100 MFU/HFU numbers have been published [here](https://github.com/mosaicml/llm-foundry/tree/main/scripts/train/benchmarking#mfu-and-hfu).
 
+Now, whenever you see MFU or HFU numbers published you have to be careful comparing those numbers to any other similar numbers, until you know that the same way was used to calculate FLOPS. Since `HFU=Actual_Achieved_FLOPS/Theoretical_FLOPS` and `Theoretical_FLOPS` are fixed. The only variable here is the `
+Actual_Achieved_FLOPS` and since most of the time an estimated value is calculated based on parameter shapes, there are many versions of calculations out there, some of which are slightly to very imprecise. Moreover you don't know how iteration time was measured.
+
+To recall `TFLOPS = FLOS / iteration_duration`:
+
+So, in order to do a fair comparison the 2 main questions to ask are:
+1. Was the total used floating point operations calculated in the same way?
+2. Was the time component calculated back-to-back of each iteration, including the `DataLoader` and logging, vs only `fwd`+`bwd` parts.
+
+If either one or both of these mismatch then you can't make a fair comparison.
+
+Unfortunately, most of the time papers and blog posts just report the MFU number w/o a link to how it was calculated.
+
+But, do not fear, most of your work will be optimizing your application rather than having a competition of who is faster. And thus as long as you consistently use whatever way you choose to calculate TFLOPS, you will immediately see when your application's performance has improved or degraded, as relative numbers are most important for you.
```

Comment:
It's not just about competition or marketing. Often you'll hear researchers/engineers say publicly "any code less than X% mfu is bad), without going into all these nuances that you aim to describe here.

So I would reword this more along the lines of:

"Do not fear, if you have trouble comparing your results with competing results keep in mind all the measurement artifacts described above that do not improve the bottom-line throughput. And thus as long as you consistently use ..."

Reply:
```suggestion
But, do not fear, if you have trouble comparing your results with competing results, remember the measurement artifacts described above.
These artifacts do not improve the bottom-line throughput, thus, as long as you consistently use whatever way you choose to calculate TFLOPS, you will immediately see when your application's performance has improved or degraded, as relative numbers are most important for you.
```

Maybe this clarifies what I intended to say